src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs

```diff
@@ -1478,6 +1454,82 @@ private static bool GetIsDaylightSavingsFromUtc(DateTime time, int year, TimeSpa
             return isDst;
         }
 
+        // This method checks if a specific year start with DST, if so, will get when the DST ends that year and return true.
+        // Otherwise will return false.
+        private static bool TryGetEndOfDstIfYearStartWithDst(int nextYear, TimeSpan utc, TimeZoneInfo zone, out DateTime dstEnd)
+        {
+            AdjustmentRule? nextYearRule = zone.GetAdjustmentRuleForTime(new DateTime(nextYear, 1, 1), out int? nextYearRuleIndex);
+
+            // DST is not supported if the year doesn't have a rule.
+            if (nextYearRule is null)
+            {
+                dstEnd = default;
+                return false;
+            }
+
+            DaylightTimeStruct nextdaylightTime;
+            // Check if the rule not specifying the DST start as Jan 1st, 12:00 AM.
+            if (!nextYearRule.IsStartDateMarkerForBeginningOfYear())
+            {
+                nextdaylightTime = zone.GetDaylightTime(nextYear, nextYearRule, nextYearRuleIndex);
+
+                // Check if we are dealing with a southern sphere time zone having a DST start date after the end date.
```

Comment:
```C#
            if (nextYearRule.IsStartDateMarkerForBeginningOfYear())
            {
                // Check if DST ends specified as Jan 1st, 12:00 AM which means the year will end with DST on.
                if (nextYearRule.IsEndDateMarkerForEndOfYear())
               {
                    dstEnd = new DateTime(nextYear, 12, 31) - utc - nextYearRule.BaseUtcOffsetDelta - nextYearRule.DaylightDelta;
                }
                else
                {
                    // Not southern sphere time zone and the year doesn't end with DST. Get the DST end date regularly as specified in the rule.
                    nextdaylightTime = zone.GetDaylightTime(nextYear, nextYearRule, nextYearRuleIndex);
                    dstEnd = nextdaylightTime.End - utc - nextYearRule.BaseUtcOffsetDelta - nextYearRule.DaylightDelta;
                 }

                 return true;
            }

                // Check if we are dealing with a southern sphere time zone.
                // If the rule specifies the DST end as of Jan 1st, 12:00 AM that means the year will end with DST on
                // but also means the year not started with DST as we already checked that before.
                if (nextdaylightTime.End < nextdaylightTime.Start && !nextYearRule.IsEndDateMarkerForEndOfYear())
                {
                    // It is a Southern sphere time zone. The year is started with DST on. Use the DST end to get the when DST ends that year.
                    dstEnd =  nextdaylightTime.End - utc - nextYearRule.BaseUtcOffsetDelta - nextYearRule.DaylightDelta;
                    return true;
                }

                dstEnd = default;
                return false;
            }
```

Reply:
This is done. Thanks for the suggestion.